middleware.go

```diff
@@ -4,22 +4,19 @@ import (
 	"net/http"
 )
 
-// NextMiddlewareFunc is the function signature for the next parameter in ServerHTTPMiddleware
-type NextMiddlewareFunc func(http.ResponseWriter, *http.Request)
```

Comment:
I think what @turbolent is saying is to keep the type as convenience, but still keep the signature of the middleware funcs as type `MiddlewareFunc func(http.ResponseWriter, *http.Request, func(http.ResponseWriter, *http.Request))`

This way, both `NextMiddlewareFunc` and `func(http.ResponseWriter, *http.Request)` will be accepted, so it is up to the user to declare if they want to pass in a wrapped `NextMiddlewareFunc` or not.